gate-core/src/main/groovy/com/netflix/spinnaker/gate/services/internal/ClouddriverService.groovy

```diff
@@ -293,6 +293,14 @@ interface ClouddriverService {
   @GET('/artifacts/credentials')
   List<Map> getArtifactCredentials()
 
+  @GET('/roles/{cloudProvider}')
+  List<Map> getRoles(@Path("cloudProvider") String cloudProvider)
+
+  @GET('/ecs/ecsclusters')
```

Comment:
I might not be completely up to date about the APIs in clouddriver but `/ecs/ecsclusters` doesn't seem quite right?

I see in another PR that the endpoint in gate would be `/ecs/cloudMetrics` but it looks like it's `/ecs/cloudmetrics` in `clouddriver`. Is there a good reason to keep it this way?

Reply:
regarding `/ecs/ecsclusters`, we use the name `ecsCluster` to distinguish between the AWS concept and the Spinnaker one, which are very different.  This is the reasoning.  It looked like the least bad solution.  It certainly is not an exciting solution :) 

The different name for the cloudMetrics endpoint is an oversight.  I'd guess that we should go lowercase all the way